### PR TITLE
gh-114364: Fix awkward wording about mmap.mmap.seekable being added

### LINKED ISSUE #114364: Awkward wording in GH-111835 whatsnew entry
# Documentation

"it requires" is a little unclear

```diff
diff --git a/Doc/whatsnew/3.13.rst b/Doc/whatsnew/3.13.rst
index 40f0cd37fe..eb00e4ad74 100644
--- a/Doc/whatsnew/3.13.rst
+++ b/Doc/whatsnew/3.13.rst
@@ -259,8 +259,8 @@ mmap
 ----

 * The :class:`mmap.mmap` class now has an :meth:`~mmap.mmap.seekable` method
-  that can be used where it requires a file-like object with seekable and
-  the :meth:`~mmap.mmap.seek` method return the new absolute position.
+  that can be used when a file-like object with seekable is required. The
+  :meth:`~mmap.mmap.seek` method now returns the new absolute position.
   (Contributed by Donghee Na and Sylvie Liberman in :gh:`111835`.)
```

I'm not sure if I should make a PR changing Doc/whatsnew/3.13.rst directly or something via Misc/NEWS.d

This refers #111852 

### Linked PRs
* gh-114374

### PR DESCRIPTION
**Pull request by technillogue**
Fixes #114364. "it requires" is a little unclear and "method return the" is incorrect.

* Issue: gh-114364

### CODE REVIEW

File: Misc/NEWS.d/3.13.0a2.rst [hunk 1]
```diff
@@ -565,9 +565,9 @@ part of a :exc:`BaseExceptionGroup`, in addition to the recent support for
 .. section: Library
 
 The :class:`mmap.mmap` class now has an :meth:`~mmap.mmap.seekable` method
-that can be used where it requires a file-like object with seekable and the
-:meth:`~mmap.mmap.seek` method return the new absolute position. Patch by
-Donghee Na.
+that can be used when a file-like object with a `seekable` method is required.
```
Comment (Eclips4) [inline Misc/NEWS.d/3.13.0a2.rst]:
```suggestion
that can be used when a file-like object with a ``seekable`` method is required.
```

File: Doc/whatsnew/3.13.rst [hunk 2]
```diff
@@ -259,8 +259,8 @@ mmap
 ----
 
 * The :class:`mmap.mmap` class now has an :meth:`~mmap.mmap.seekable` method
-  that can be used where it requires a file-like object with seekable and
-  the :meth:`~mmap.mmap.seek` method return the new absolute position.
+  that can be used when a file-like object with a `seekable` method is required.
```
Comment (Eclips4) [inline Doc/whatsnew/3.13.rst]:
```suggestion
  that can be used when a file-like object with a ``seekable`` method is required.
```

**Comment by corona10** [comment]
@AlexWaygood Can you please take a look?

File: Doc/whatsnew/3.13.rst [hunk 3]
```diff
@@ -259,8 +259,8 @@ mmap
 ----
 
 * The :class:`mmap.mmap` class now has an :meth:`~mmap.mmap.seekable` method
-  that can be used where it requires a file-like object with seekable and
-  the :meth:`~mmap.mmap.seek` method return the new absolute position.
+  that can be used when a file-like object with a ``seekable`` method is required.
```
Comment (erlend-aasland) [inline Doc/whatsnew/3.13.rst]:
```suggestion
  that can be used when a seekable file-like object is required.
```

File: Misc/NEWS.d/3.13.0a2.rst [hunk 4]
```diff
@@ -565,9 +565,9 @@ part of a :exc:`BaseExceptionGroup`, in addition to the recent support for
 .. section: Library
 
 The :class:`mmap.mmap` class now has an :meth:`~mmap.mmap.seekable` method
-that can be used where it requires a file-like object with seekable and the
-:meth:`~mmap.mmap.seek` method return the new absolute position. Patch by
-Donghee Na.
+that can be used when a file-like object with a ``seekable`` method is required.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/3.13.0a2.rst]:
```suggestion
that can be used when a seekable file-like object is required.
```

**Review by erlend-aasland — COMMENTED** [review]
I suggested @ericvsmith's wording from the issue.

File: Doc/whatsnew/3.13.rst (hunk 3 above)
Comment (technillogue) [reply Doc/whatsnew/3.13.rst]: sure, my rationale is that mmap was already seekable (you could call seek()), it's ZipFile that checks for seekable() specifically, but I'm also fine with this wording

Comment (erlend-aasland) [reply Doc/whatsnew/3.13.rst]: @corona10?

**Pull request merged**